src/Sylius/Bundle/PaymentBundle/Form/Type/PaymentGatewayChoiceType.php

```diff
@@ -11,31 +11,27 @@
 
 namespace Sylius\Bundle\PaymentBundle\Form\Type;
 
+use Payum\Core\Registry\RegistryInterface;
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
- * Payment gateway choice type.
- *
  * @author Paweł Jędrzejewski <pawel@sylius.org>
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
  */
 class PaymentGatewayChoiceType extends AbstractType
 {
     /**
-     * Choices.
-     *
-     * @var array
+     * @var RegistryInterface
      */
-    protected $gateways;
+    protected $gatewaysRegistry;
 
     /**
-     * Constructor.
-     *
-     * @param array $gateways
+     * @param RegistryInterface $gatewaysRegistry
      */
-    public function __construct(array $gateways)
+    public function __construct(RegistryInterface $gatewaysRegistry)
```

Comment:
Would be better to use GatewayRegistryInterface https://github.com/Payum/Payum/blob/master/src/Payum/Core/Registry/GatewayRegistryInterface.php